src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,105 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final int DEFAULT_CAPACITY = 10;
+    private int size;
+    private T[] array;
+
+    public ArrayList() {
+        this.array = (T[]) new Object[DEFAULT_CAPACITY];
+    }
+
     @Override
     public void add(T value) {
-
+        if (size >= array.length) {
+            grow(size);
```

Comment:
```suggestion
            grow();
```
size is available in private method because it is a field